eng/pipelines/templates/stages/archetype-sdk-tool-azure-function.yml

```diff
@@ -0,0 +1,93 @@
+parameters:
+- name: ToolName
+  type: string
+- name: FunctionProject
+  type: string
+- name: TestProject
+  type: string
+- name: ProductionEnvironmentName
+  type: string
+- name: StagingEnvironmentName
+  type: string
+
+stages:
+    - stage: Build
+      variables:
+      - template: ../variables/globals.yml
+      jobs:
+      - job: BuildPackage
+        pool:
+          vmImage: ubuntu-16.04
+        steps:
+          - task: UseDotNet@2
+            displayName: Tools
+            inputs:
+              version: '3.1.201'
+          - pwsh: |
+              dotnet restore tools/${{ parameters.ToolName }}
+            displayName: Restore
+          - pwsh: |
+              dotnet build tools/${{ parameters.ToolName }} --configuration Release --no-restore
+            displayName: Build
+          - pwsh: |
+              dotnet test tools/${{ parameters.ToolName }}/${{ parameters.TestProject }} --configuration Release
+            displayName: Test
+          - pwsh: |
+              dotnet publish tools/${{ parameters.ToolName }}/${{ parameters.FunctionProject }} --configuration Release --no-restore --output $(Build.ArtifactStagingDirectory)/app
+            displayName: Publish
+          - publish: $(Build.ArtifactStagingDirectory)
+            displayName: Upload
+            artifact: artifacts
+            condition: succeededOrFailed()
+
+    - ${{if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal'))}}:
+      - stage: Staging
+        dependsOn: Build
+        jobs:
+        - deployment: PublishFunction
+          environment: ${{ parameters.StagingEnvironmentName }}
+          pool:
+            vmImage: ubuntu-16.04
+          strategy:
+            runOnce:
+              deploy:
+                steps:
+                - task: ArchiveFiles@2
+                  displayName: Archive
+                  inputs:
+                    rootFolderOrFile: $(Pipeline.Workspace)/artifacts/app
+                    includeRootFolder: false
+                    archiveFile: $(Pipeline.Workspace)/artifacts/${{ parameters.ToolName }}-$(Build.BuildID).zip
+                - task: AzureFunctionApp@1
```

Comment:
Shouldn't we also sign these?

Reply:
I'm looking into this. We can probably sign it but need to think through whether there is a benefit. We aren't publishing the artifacts anywhere, they are running in production and whilst it is nice for them to be signed I'm not sure Azure Functions can enforce code integrity based on code signing.

Looking into it with the team that owns some of that tooling.